src/storage/src/hummock/iterator/concat_inner.rs

```diff
@@ -198,14 +147,14 @@ impl<TI: SstableIteratorType> HummockIterator for ConcatIteratorInner<TI> {
                 .tables
                 .partition_point(|table| match Self::Direction::direction() {
                     DirectionEnum::Forward => {
-                        let ord = FullKey::decode(table.smallest_key()).cmp(&key);
+                        let ord = FullKey::decode(smallest_key(table)).cmp(&key);
 
                         ord == Less || ord == Equal
                     }
                     DirectionEnum::Backward => {
-                        let ord = FullKey::decode(table.largest_key()).cmp(&key);
-
-                        ord == Greater || ord == Equal
+                        let ord = FullKey::decode(largest_key(table)).cmp(&key);
+                        ord == Greater
+                            || (ord == Equal && !table.key_range.as_ref().unwrap().right_exclusive)
```

Comment:
This logic is easy to use wrongly, encapsulate it into a function or add more comments to explain it?